File: core/src/main/java/org/adoptopenjdk/jheappo/objects/RootNativeStack.java
package org.adoptopenjdk.jheappo.objects;

/*
 * Copyright (c) 2018 Kirk Pepperdine.
 * Licensed under https://github.com/AdoptOpenJDK/jheappo/blob/master/LICENSE
 * Instructions: https://github.com/AdoptOpenJDK/jheappo/wiki
 */


import org.adoptopenjdk.jheappo.io.EncodedChunk;

public class RootNativeStack extends HeapObject {

    public final static int TAG = 0x04;

    private int threadSerialNumber;

    public RootNativeStack(EncodedChunk buffer) {
        super(buffer);
        threadSerialNumber = buffer.extractU4();
    }
}

File: src/main/java/org/adoptopenjdk/jheappo/objects/RootNativeStack.kt
package org.adoptopenjdk.jheappo.objects

/*
 * Copyright (c) 2018 Kirk Pepperdine.
 * Licensed under https://github.com/AdoptOpenJDK/jheappo/blob/master/LICENSE
 * Instructions: https://github.com/AdoptOpenJDK/jheappo/wiki
 */

import org.adoptopenjdk.jheappo.heapdump.EncodedChunk

class RootNativeStack(buffer: EncodedChunk) : HeapObject(buffer) {

    companion object {
        const val TAG = 0x04
    }

    private val threadSerialNumber: Int = buffer.extractU4()
}